tests/e2e/tx/service_test.go

```diff
@@ -908,6 +908,148 @@ func (s IntegrationTestSuite) TestTxDecode_GRPCGateway() {
 	}
 }
 
+func (s IntegrationTestSuite) TestTxEncodeAmino_GRPC() {
+	val := s.network.Validators[0]
+	txBuilder := s.mkTxBuilder()
+	stdTx, err := clienttx.ConvertTxToStdTx(val.ClientCtx.LegacyAmino, txBuilder.GetTx())
+	s.Require().NoError(err)
+	txJSONBytes, err := val.ClientCtx.LegacyAmino.MarshalJSON(stdTx)
+	s.Require().NoError(err)
+
+	testCases := []struct {
+		name      string
+		req       *tx.TxEncodeAminoRequest
+		expErr    bool
+		expErrMsg string
+	}{
+		{"nil request", nil, true, "request cannot be nil"},
+		{"empty request", &tx.TxEncodeAminoRequest{}, true, "invalid empty tx json"},
+		{"valid request with amino-json", &tx.TxEncodeAminoRequest{AminoJson: string(txJSONBytes)}, false, ""},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		s.Run(tc.name, func() {
+			res, err := s.queryClient.TxEncodeAmino(context.Background(), tc.req)
+			if tc.expErr {
+				s.Require().Error(err)
+				s.Require().Contains(err.Error(), tc.expErrMsg)
+				s.Require().Empty(res)
+			} else {
+				s.Require().NoError(err)
+				s.Require().NotEmpty(res.GetAminoBinary())
```

Comment:
To make sure the test is correct, let's add an additional check here:

```go
newStdTx, err := val.ClientCtx.TxConfig.TxDecoder()(res.GetAminoBinary())
```

and make sure stdTx and newStdTx are equal

(Same for other endpoints)